CHANGELOG.md

```diff
@@ -2,6 +2,8 @@
 
 - Always break up TYPEOF SOQL queries ([issue](https://github.com/dangmai/prettier-plugin-apex/issues/244)).
 
+- Fixes weird `new List<Object>(0)` constructor with the more usual `new Object[0]` syntax when possible. Also adds new tests that cover edge cases. 
```

Comment:
Feel free to credit yourself here, e.g. `(credit to @brianmfear)` at the end of this line. I'll add it after merge anyway :)